notebooks/COS/ViewData/ViewData.ipynb

```diff
@@ -0,0 +1,1452 @@
+    "    ax0.legend(fontsize = 20)\n",
+    "    ax0.set_yscale('log')\n",
+    "    ax0.set_title(\"Fig. 3.1\\nData quality flag frequencies\",fontsize = 20)\n",
+    "\n",
+    "for i in range(3): # Next loop through the 3 NUV segments/stripes to make bottom plot\n",
+    "    dataQual, segment = nuv_x1d_data[i][\"DQ\"], nuv_x1d_data[i][\"SEGMENT\"] # Gather the data\n",
+    "    bins_ = np.arange(-1,15) # all the log2 of the DQ values\n",
+    "\n",
+    "    logDQ = np.log2(dataQual)\n",
+    "    logDQ[np.where(-1*(logDQ) == np.inf)] = -1\n",
+    "\n",
+    "    ax1.hist(logDQ , bins = bins_, color = 'cmy'[i], alpha = 1 - 0.25*(i) , label = segment, hatch = '|-/'[i]) # Add hatching to distinguish overlapping colors\n",
+    "    ax1.legend(fontsize = 20)\n",
+    "    ax1.set_yscale('log')\n",
+    "    \n",
+    "plt.xticks(bins_+0.5, labels = meanings2, rotation = 'vertical', fontsize = 15)\n",
+    "fig.set_tight_layout('tight')\n",
+    "plt.savefig(plotsdir+'Fig3.1.png', dpi = 200)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "When the data is processed through `calCOS`, it has a summed set of \"Serious Data Quality\" [SDQ](https://hst-docs.stsci.edu/cosdhb/chapter-3-cos-calibration/3-4-descriptions-of-spectroscopic-calibration-steps) flags attached to it in the form of a single integer between 0 and $2^{15}$. This value is found in the 1-th header as `\"SDQFLAGS\"`. This can be decomposed into the powers of 2 data flags above, (by showing its binary representation,) and tells the pipeline whether to weight that pixel in that exposure by 0 or 1. These weights are then stored in the `x1d` and `x1dsum` files, in the `\"DQ_WGT\"` column. The `\"DQ_WGT\"` value for a given pixel in the `x1dsum` file will then be equal to the number of exposures which went into creating that `x1dsum` file in which no serious data quality flags apply to that pixel.\n",
+    "\n",
+    "Below, we first see *which* data quality flags the pipeline considered \"serious\", and then demonstrate masking out the datapoints assigned a weight of 0."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sdqFlags_fuv = fits.getheader(fuv_x1d_filepath,1)[\"SDQFLAGS\"]\n",
+    "print(f\"The FUV was processed with SDQFLAGS = {sdqFlags_fuv}, which in binary is {bin(sdqFlags_fuv)[2:]}\")\n",
+    "print(\"\\t\\tThus, the following DQ-flagged data get weighted by 0:\\n\")\n",
+    "for i, char in enumerate(bin(sdqFlags_fuv)[2:]):\n",
+    "    if char =='1':\n",
+    "        print(f\"\\t\\t\\t{(2**(i+1))}'s place - \", meanings[i+2])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mask_noSDQ = np.ndarray.astype(fuv_x1d_data[0][\"DQ_WGT\"], bool)\n",
```

Comment:
```suggestion
    "mask_noSDQ = fuv_x1d_data[0][\"DQ_WGT\"].astype(bool)\n",
```
this is the more idiomatic way to do ``astype`` unless it doesn't work for some reason.